rpc-server/src/dispatchers/blockchain.rs

```diff
@@ -71,16 +71,16 @@ fn get_validator_by_address(
         let staker_addresses =
             StakingContract::get_validator_stakers(accounts_tree, &db_txn, address);
 
-        let mut stakers_map = HashMap::new();
+        let mut stakers_list: Vec<Staker> = vec![];
 
         for address in staker_addresses {
             let staker = StakingContract::get_staker(accounts_tree, &db_txn, &address).unwrap();
             if !staker.balance.is_zero() {
-                stakers_map.insert(address, staker.balance);
+                stakers_list.push(Staker::from_staker(&staker));
```

Comment:
You could set the delegation to None directly before this line.

Actually it worth to check if a staker's balance can actually be zero. I think that check became unnecessary.

Reply:
@sisou thank you for the review. I've set the `delegation` to `None`. I wasn't able to verify whether the protocol allows it to have a delegated stake with a balance of zero. Maybe @ii-cruz can verify this since he wrote the initial code.

Reply:
I've given it a look and so did Pascal, this check doesn't seem to be needed indeed. :) We are removing the staker if the balance drops to zero :)